File: app/src/main/java/com/letianpai/robot/desktop/utils/ImageDownloader.java
package com.letianpai.robot.desktop.utils;

import java.io.File;
import java.io.FileOutputStream;
import java.io.IOException;
import java.io.InputStream;
import java.net.URL;

import okhttp3.OkHttpClient;
import okhttp3.Request;
import okhttp3.Response;

import java.io.File;
import java.io.FileOutputStream;
import java.io.IOException;
import java.io.InputStream;
import okhttp3.OkHttpClient;
import okhttp3.Request;
import okhttp3.Response;

public class ImageDownloader {

    public static String downloadImage(String urlString, String destinationPath) {
        OkHttpClient client = new OkHttpClient();
        Request request = new Request.Builder()
                .url(urlString)
                .build();

        try (Response response = client.newCall(request).execute()) {
            if (response.isSuccessful() && response.body() != null) {
                InputStream inputStream = response.body().byteStream();
                File destinationFile = new File(destinationPath);
                FileOutputStream outputStream = new FileOutputStream(destinationFile);

                byte[] buffer = new byte[1024];
                int bytesRead;

                while ((bytesRead = inputStream.read(buffer)) != -1) {
                    outputStream.write(buffer, 0, bytesRead);
                }

                outputStream.close();
                inputStream.close();

                return destinationPath;
            } else {
                return null;
            }
        } catch (IOException e) {
            e.printStackTrace();
            return null;
        }
    }
}

File: app/src/main/java/com/letianpai/robot/desktop/utils/ImageDownloader.kt
package com.letianpai.robot.desktop.utils

import okhttp3.OkHttpClient
import okhttp3.Request
import okhttp3.Request.Builder
import java.io.File
import java.io.FileOutputStream
import java.io.IOException

object ImageDownloader {
    fun downloadImage(urlString: String, destinationPath: String): String? {
        val client = OkHttpClient()
        val request: Request = Builder()
            .url(urlString)
            .build()

        try {
            client.newCall(request).execute().use { response ->
                if (response.isSuccessful && response.body != null) {
                    val inputStream = response.body!!.byteStream()
                    val destinationFile = File(destinationPath)
                    val outputStream = FileOutputStream(destinationFile)

                    val buffer = ByteArray(1024)
                    var bytesRead: Int

                    while ((inputStream.read(buffer).also { bytesRead = it }) != -1) {
                        outputStream.write(buffer, 0, bytesRead)
                    }

                    outputStream.close()
                    inputStream.close()

                    return destinationPath
                } else {
                    return null
                }
            }
        } catch (e: IOException) {
            e.printStackTrace()
            return null
        }
    }
}
